core/templates/components/state-directives/response-header/response-header.component.ts

```diff
@@ -16,80 +16,76 @@
  * @fileoverview Component for the header of the response tiles.
  */
 
-require('domain/utilities/url-interpolation.service.ts');
-require(
-  'components/state-editor/state-editor-properties-services/' +
-  'state-editor.service.ts');
-require(
-  'components/state-editor/state-editor-properties-services/' +
-  'state-interaction-id.service');
-require(
-  'components/state-editor/state-editor-properties-services/' +
-  'state-property.service.ts');
-require('services/editability.service.ts');
+import { Component, Input } from '@angular/core';
+import { StateEditorService } from 'components/state-editor/state-editor-properties-services/state-editor.service';
+import { StateInteractionIdService } from 'components/state-editor/state-editor-properties-services/state-interaction-id.service';
+import { EditabilityService } from 'services/editability.service';
+import INTERACTION_SPECS from 'interactions/interaction_specs.json';
+import { AppConstants } from 'app.constants';
+import { downgradeComponent } from '@angular/upgrade/static';
 
-angular.module('oppia').component('responseHeader', {
-  bindings: {
-    getIndex: '&index',
-    getOutcome: '&outcome',
-    getSummary: '&summary',
-    getShortSummary: '&shortSummary',
-    isActive: '&isActive',
-    getOnDeleteFn: '&onDeleteFn',
-    getNumRules: '&numRules',
-    isResponse: '&isResponse',
-    showWarning: '&showWarning',
-    navigateToState: '='
-  },
-  template: require(
-    'components/state-directives/response-header/' +
-    'response-header.component.html'),
-  controllerAs: '$ctrl',
-  controller: [
-    'EditabilityService', 'StateEditorService', 'StateInteractionIdService',
-    'INTERACTION_SPECS', 'PLACEHOLDER_OUTCOME_DEST',
-    function(
-        EditabilityService, StateEditorService, StateInteractionIdService,
-        INTERACTION_SPECS, PLACEHOLDER_OUTCOME_DEST) {
-      var ctrl = this;
-      ctrl.isInQuestionMode = function() {
-        return StateEditorService.isInQuestionMode();
-      };
+@Component({
+  selector: 'oppia-response-header',
+  templateUrl: './response-header.component.html'
+})
+export class ResponseHeaderComponent {
+  @Input() index;
+  @Input() summary;
+  @Input() shortSummary;
+  @Input() isActive;
+  @Input() deleteFn;
+  @Input() outcome;
```

Comment:
We try to remove the pattern of passing down functions from parent components to child. Instead, use @Output here and listen to that event in parent component and call the original function in the parent component itself.
Here and elsewhere.

Reply:
Done